src/ffi.rs

```diff
@@ -0,0 +1,159 @@
+use crate::embed;
+use std::ffi::{CStr, CString};
+use std::os::raw::c_char;
+use std::ptr::null_mut;
+use std::sync::Once;
+
+static INIT: Once = Once::new();
+
+pub fn init_log() {
+    INIT.call_once(env_logger::init);
+}
+
+#[no_mangle]
+pub extern "C" fn dispatch(
+    db_name: *const c_char,
+    rpc: *const c_char,
+    args: *const c_char,
```

Comment:
Do we need to worry about embedded nulls in the args? I feel like we do.

Reply:
By worry, do you mean "don't crash if present", or "retain through all transformations"? I think we expect the first two arguments to be regular strings and the third to be utf-8 encoded JSON, so I presume it's just the former, right?

I think that's covered through the bad encoding tests like https://github.com/rocicorp/repc/pull/104/files#diff-49a4b4d69f811d5619c481802227b86eR127.

Reply:
I mean retain so that we can properly support client expectations.

I forgot that utf8 doesn't have nulls in it. I think all three should be defined to be utf8, so hooray.